### DISCUSSION TITLE
What is going on with the Function ID ?

### DISCUSSION THREAD

**Discussion by Sam115200** [Shopify/function-examples#298] [Feedback: Delivery Customization API]
This applies to all functions, not only delivery customizations.

Previously, function IDs had different values and now the value is changed to a new value all of a sudden, without any notice or change log. Causing multiple issues in the app experience. 

For example, this was the value for function ID for this function, `01H23FJ766XZAFYS2B0Q68MFXQ` and now for the same function it shows up to be `2bff928e-972d-40f8-ac79-4d6c1c7473e2` as shown below.
<img width="973" alt="Screenshot 2023-08-02 at 5 54 58 PM" src="https://github.com/Shopify/function-examples/assets/139220490/ce7204bf-6e24-4d5c-9f31-76e6caa27649">

If we create new function, using the new function ID then it works fine. But trying to use the new function ID generates errors with already created functions, for example, trying to update the function status or metafield value causes this graphql API error,
```
Error: Function ID cannot be changed.
```

So, it leaves the app in a weird state, do we use the UUID based new function IDs moving forward? then how to deal with already existing thousands of functions which can't be updated using the new function IDs?

If this kind of big change is aggressively being rolled out which can break apps which use functions, why is this not communicated to developers more clearly before being rolled out? Not even a warning?

**Comment by davejcameron** [comment]
Hey @Sam115200 as part of the work to support Simplified deployments and allow Functions to have versions, another identifier was introduced in this UUID format. The intention was that either identifiers would work seamlessly together. This is a bug that it's failing with that message and we'll get that fixed.

**Reply by Sam115200** [reply]
@davejcameron thank you for the info, it is helpful. So, the goal is, we will only be using the new function IDs as a drop in replacement even for the existing functions and it should work. If possible, please let me know once the bug is fixed so I can do some testing.

**Reply by DuncanUszkay1** [reply]
This bug should be resolved now for payment customizations 🙏🏻 Thanks for your report!

**Reply by Sam115200** [reply]
I've tested it a few times and it seems to work now and I do not see any issue. Thank you.

**Reply by Sam115200** [reply]
@DuncanUszkay1 Unfortunately, it seems there are still intermittent issues happening. Merchant reported to me that he got this error while trying to create a new function,
```
Could not find Function with id: 01H0WTM97396DYED1SEDA5VW6S
```
Where `01H0WTM97396DYED1SEDA5VW6S` is the function ID in the old format. I thought new and old both function identifiers should be able to work seamlessly together, or is my assumption wrong and we have to use the new function IDs to create new functions?

The merchant was able to create the function on his second attempt, so this is intermittent issue which only happens sometimes.

**Comment by Sam115200** [comment]
I can confirm that the issue of `Could not find Function with id` has been resolved in the last 2 weeks or so. Before it kept happening randomly, but now it has completely stopped. 

The issue is resolved now, closing this thread. Thank you team for all the hard work!
